engine/classes/ElggAutoP.php

```diff
@@ -108,7 +110,19 @@ public function process($html) {
 		$this->_xpath = new DOMXPath($this->_doc);
 		// start processing recursively at the BODY element
 		$nodeList = $this->_xpath->query('//body[1]');
-		$this->addParagraphs($nodeList->item(0));
+		$body = $nodeList->item(0);
+		if (!$body instanceof DOMElement && $this->debug) {
```

Comment:
Could you add some comments that describe what is being done in the ```if``` block and why?